util/vendor.py

```diff
@@ -340,46 +358,82 @@ def main(argv):
         raise SystemExit(sys.exc_info()[1])
     desc['_base_dir'] = vendor_file_base_dir
 
+    first_import = False
+
     # Load lock file contents (if possible)
-    desc_file_path_str = str(desc_file_path)
-    lock_file_path = Path(
-        desc_file_path_str[:desc_file_path_str.find('.vendor.hjson')] +
-        '.lock.hjson')
+    desc_file_stem = desc_file_path.name.rsplit('.', 2)[0]
+    lock_file_path = desc_file_path.with_name(desc_file_stem + '.lock.hjson')
     try:
-        with open(lock_file_path, 'r') as f:
+        with open(str(lock_file_path), 'r') as f:
             lock = hjson.loads(f.read(), use_decimal=True)
     except FileNotFoundError:
-        log.warning(
-            "Unable to read lock file %s. Assuming this is the first import.",
-            lock_file_path)
         lock = None
+        if args.update:
+            log.fatal("Cannot update without existing lock file %s.",
+                      str(lock_file_path))
+            log.fatal(
+                "Retry without the '--update' flag if this is an initial import."
+            )
+            raise SystemExit(1)
+        else:
+            log.warning(
+                "Unable to read lock file %s. Assuming this is the first import.",
+                str(lock_file_path))
+            first_import = True
```

Comment:
You can simplify the logic with "`--update` is implied if no lock file exists".

```
ignore_lock_file = args.update or log_file_doesnt_exist
```

and then continue as usual

Reply:
This is a little more complex, because `first_import` was used by `get_shortlog`. I've tried to match the suggested logic, along with the recommendation about `import_desc` below, but one of the compilations is that the `lock` information is needed when finding out what changed.

The end result is the logic will need re-reviewing, because it is simpler (update is now assumed if the lock file does not exist), but it has also changed quite a bit.